system_metrics/system_metrics_client.go

```diff
@@ -96,15 +96,16 @@ type ConnectionOptions struct {
 }
 
 func (opts ConnectionOptions) validate() error {
-	if (opts.DialOpts.APIKey == "" && opts.DialOpts.Username != "") ||
-		(opts.DialOpts.APIKey != "" && opts.DialOpts.Username == "") {
+	hasAuth := opts.DialOpts.APIKey != "" || opts.DialOpts.Password != ""
```

Comment:
you can just do `hasAuth := opts.DialOpts.Username`. If there is a username there has to be password or API key.